src/decode_metar.c

```diff
@@ -2223,7 +2231,57 @@ static MDSP_BOOL isTimeUTC( char *UTC, Decoded_METAR *Mptr, int *NDEX )
    else
       return FALSE;
 }
- 
+
+#pragma subtitle(" ")
+#pragma page(1)
+#pragma subtitle("subtitle - description                       ")
+/********************************************************************/
+/*                                                                  */
+/*  Title:         defaultWindUnitsForAirport                       */
+/*  Organization:  W/OSO242 - GRAPHICS AND DISPLAY SECTION          */
+/*  Date:          15 Sep 1994                                      */
+/*  Programmer:    CARL MCCALLA                                     */
+/*  Language:      C/370                                            */
+/*                                                                  */
+/*  Abstract:                                                       */
+/*                                                                  */
+/*  External Functions Called:                                      */
+/*                 None.                                            */
+/*                                                                  */
+/*  Input:         x                                                */
+/*                                                                  */
+/*  Output:        x                                                */
+/*                                                                  */
+/*  Modification History:                                           */
+/*                 None.                                            */
+/*                                                                  */
+/********************************************************************/
+#pragma page(1)
+
+static MDSP_BOOL defaultWindUnitsForAirport( char *airportId, char **defaultUnit )
+{
+   int i;
+   static char *airportPrefixUsingMPS[1] = { "Z" },
+               *airportPrefixWithoutDefault[1] = { "U" };
+
+   char airportPrefix = airportId[0];
```

Comment:
might be good to also verify that the strlen(airportId) is 4, to guard against non-ICAO airport codes getting passed in somehow.